src/pages/LogIn.jsx

```diff
@@ -1,65 +1,126 @@
+      setPassword('');
+
+    } catch (err) {
+      if (!err?.response) {
+        setErrMsg('No Server Response');
+      } else if (err.response?.status === 401) {
+        setErrMsg('Invalid Credentials');
+      } else {
+        setErrMsg('Login Failed');
+      }
+    }
+  };
+
+  // Handle Logout
+  const handleLogout = () => {
+    // Remove token from local storage
+    localStorage.removeItem('token');
+    setIsLoggedIn(false); // Update login state
+    window.location.reload(); // Optionally refresh the page
+  };
 
-const handleSubmit = async (e) => {
-  e.preventDefault();
-  console.log(user, pwd);
-  setUser('');
-  setPwd('');
-}
   return (
-  <>
-  <section>
-    <div className="container mx-auto max-w-3xl px-4 py-8">
-      <h1 className="text-3xl font-bold mb-6">Sign In</h1>
+    <>
+      <section>
+        <div className="container mx-auto max-w-3xl px-4 py-8">
+          <h1 className="text-3xl font-bold mb-6">Sign In</h1>
           <p ref={errRef} className={errMsg ? "errmsg" : "offscreen"} aria-live="assertive">{errMsg}</p>
-        <form onSubmit={handleSubmit}>
-          <label htmlFor="username">Username:</label>
-          <input 
-            type="text" 
-            id="username"
-            ref={userRef}
-            autoComplete="off"
-            onChange={(e) => setUser(e.target.value)}
-            value={user}
-            required
-          />
-          <label htmlFor="password">Password:</label>
-          <input 
-            type="password" 
-            id="password"
-            onChange={(e) => setPwd(e.target.value)}
-            value={pwd}
-            required
-          />
-          <button>Sign In</button>
-        </form>
-        <p>
-          Need an Account?<br />
-          <span className="line">
-            {/*put router link here*/}
-            <a href="#">Sign Up</a>
-          </span>
-        </p>
-      </div>
-    </section>
-</>
- )
+
+          {/* If logged in, show the Logout button, otherwise show login form */}
+          {isLoggedIn ? (
+            <div>
+              <p>You are logged in!</p>
+              <button onClick={handleLogout}>Logout</button>
+            </div>
+          ) : (
+            <form onSubmit={handleSubmit}>
+              <label htmlFor="email">Email:</label>
+              <input
+                type="email"
+                id="email"
+                ref={userRef}
+                autoComplete="off"
+                onChange={(e) => setEmail(e.target.value)}
+                value={email}
+                required
+              />
+
+              <label htmlFor="password">Password:</label>
+              <input
+                type="password"
+                id="password"
+                onChange={(e) => setPassword(e.target.value)}
+                value={password}
+                required
+              />
+
+              {/* Disable button if email is invalid */}
+              <button disabled={!validEmail}>
```

Comment:
This is throwing an error since `validEmail` is no longer defined. Would change this disabled state based on whether the two input field are empty. If both have some value filled in, then the button can be enabled

Reply:
Yes. Missed it. Corrected.